src/server/enterprise/server/api_server.go

```diff
@@ -332,3 +346,237 @@ func (a *apiServer) Deactivate(ctx context.Context, req *ec.DeactivateRequest) (
+				Replicas: 0,
+			},
+		}, metav1.UpdateOptions{
+			FieldManager: "enterprise-server",
+		}); err != nil {
+			return errors.Errorf("could not scale down %s: %v", w.GetName(), err)
+		}
+	}
+	return nil
+}
+
+func (a *apiServer) Unpause(ctx context.Context, req *ec.UnpauseRequest) (resp *ec.UnpauseResponse, retErr error) {
+	if a.env.Mode == SidecarMode || a.env.Mode == EnterpriseMode {
+		return nil, errors.Errorf("cannot pause a sidecar or enterprise server")
+	}
+	if err := a.rollPachd(ctx, false); err != nil {
+		return nil, errors.EnsureStack(err)
+	}
+
+	return &ec.UnpauseResponse{}, nil
+}
+
+// PauseStatus checks the pause status of a cluster.  It does this by checking
+// first for the pachd-config ConfigMap; if it does not exist, then the cluster
+// is by default unpaused.  Next it looks for the “pachyderm.com/updatedAt”
+// field; if that field does not exist then this must be an empty config map and
+// the cluster is by default paused.  If it does exist, though, it is remembered.
+//
+// Finally it cycles through all pachd pods, checking that they were created
+// before the updateAt value; if some are before & some after (or at the exact
+// same instant), then the cluster is considered partially paused; if all pachd
+// pods were created before the ConfigMap was updated then it has not taken
+// effect and the cluster is in the reverse state; if all pods were created
+// after the ConfigMap was updated then it has taken effect and the cluster is
+// in the indicated state.
+func (a *apiServer) PauseStatus(ctx context.Context, req *ec.PauseStatusRequest) (resp *ec.PauseStatusResponse, retErr error) {
+	kc := a.env.GetKubeClient()
+	cc := kc.CoreV1().ConfigMaps(a.env.Namespace)
+	c, err := cc.Get(ctx, "pachd-config", metav1.GetOptions{})
+	if k8serrors.IsNotFound(err) {
+		// If there is no configmap, then the pachd pods must be
+		// unpaused.
+		return &ec.PauseStatusResponse{
+			Status: ec.PauseStatusResponse_UNPAUSED,
+		}, nil
+	} else if err != nil {
+		return nil, errors.Errorf("could not get configmap: %v", err)
+	}
+	if c.Annotations[updatedAtFieldName] == "" {
+		// If there is no updatedAt, then then the pachd pods must be in
+		// their default, unpaused, state.
+		return &ec.PauseStatusResponse{
+			Status: ec.PauseStatusResponse_UNPAUSED,
+		}, nil
+	}
+	updatedAt, err := time.Parse(time.RFC3339, c.Annotations[updatedAtFieldName])
+	if err != nil {
+		return nil, errors.Errorf("could not parse update time %v: %v", c.Annotations["pachyderm.com/updatedAt"], err)
+	}
+
+	pods := kc.CoreV1().Pods(a.env.Namespace)
+	pp, err := pods.List(ctx, metav1.ListOptions{
+		LabelSelector: "app=pachd",
+	})
+	if err != nil {
+		return nil, errors.Errorf("could not get pachd pods: %v", err)
+	}
+	var sawAfter, sawBefore bool
+	for _, p := range pp.Items {
+		if p.CreationTimestamp.Time.Before(updatedAt) {
+			sawBefore = true
+		} else {
+			sawAfter = true
+		}
+	}
+	switch {
+	case sawBefore && sawAfter:
```

Comment:
nit: I think setting the correct value of `PauseStatus` in the switch statement and handling return after would save a bunch of lines of code. Up to you